models/Version/Adapter/DatabaseVersionStorageAdapter.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Version\Adapter;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Result;
+use Doctrine\DBAL\ParameterType;
+
+class DatabaseVersionStorageAdapter implements VersionStorageAdapterInterface
+{
+    CONST versionsTableName = "versionsData";
+
+    protected function getDb() : Connection
+    {
+        return \Pimcore::getContainer()->get($this->database_connection);
+    }
+
+    public function __construct(protected string $database_connection)
```

Comment:
```suggestion
    public function __construct(protected ConnectionInterface $databaseConnection)
```
The Doctrine Bundle registers all configured connections automatically as a service (using the following scheme `doctrine.dbal.[name]_connection`), so we can inject the service directly rather than using the service name.